### PR TITLE
Fix: improves stability on get_available_memory

### PR DESCRIPTION
**Pull request by RuffaloLavoisier**
**What**:
1. Added safety to check if it is null before using the file pointer.
2. Check buf null pointer, change fread() condition check.

**Why**:
1. File pointer may be null due to environmental factors.
2. If the fread fails, the buffer may be NULL.

**How**:
1. Add check condition for file pointer.
2. Add check condition for buf

**Checklist**:

- [ ] Tests
- [ ] PR merge commit message adjusted

### CODE REVIEW

File: src/pluginlaunch.c [hunk 1]
```diff
@@ -398,14 +398,20 @@ plugin_timeout (nvti_t *nvti)
...
   fd = fopen ("/proc/meminfo", "r");
+  if (fd == NULL)
+    {
+      g_warning ("Couldn't open /proc/meminfo");
+      return 0;
+    }
   len = fread (buf, 1, sizeof (buf) - 1, fd);
   fclose (fd);
-  if (len == 0)
+  if (len < (sizeof (buf) - 1) || buf == NULL)
```
Comment (jjnicola) [inline src/pluginlaunch.c:414]: If the content of the file has fewer chars than the buffer size, should not be an error.
Actually, I have just checked with `wc -c /proc/meminfo` and I got 1503 chars.

**Review by jjnicola — CHANGES_REQUESTED** [review]

**Comment by jjnicola** [comment]
Hi @RuffaloLavoisier ! thanks for your collaboration. Please, take a look at https://github.com/greenbone/openvas-scanner/tree/main/RELICENSE

File: src/pluginlaunch.c (hunk 1 above)
Comment (RuffaloLavoisier) [reply src/pluginlaunch.c:414]: Hi, @jjnicola 
Thank you for review.

Oh I see, i ii revert it.

haha thanks 👍 
> Actually, I have just checked with wc -c /proc/meminfo and I got 1503 chars.

**Review by jjnicola — APPROVED** [review]

**Comment by RuffaloLavoisier** [comment]
@jjnicola 

Hi there
When will it be merged into the main branch?

**Review by nichtsfrei — APPROVED** [review]

**Pull request merged**